generate_analytical_dataset.py

```diff
@@ -0,0 +1,274 @@
+        replace_element_value_wit_key_value_pair(db_object, '_removedDateTime', '$date', formatted_removed_date_time_as_string)
+
+    archived_date_time_as_string = retrieve_date_time_element('_archivedDateTime', db_object)
+    if archived_date_time_as_string:
+        formatted_archived_date_time_as_string = format_date_to_valid_outgoing_format(archived_date_time_as_string)
+        replace_element_value_wit_key_value_pair(db_object, '_archivedDateTime', '$date', formatted_archived_date_time_as_string)
+    return db_object
+
+def retrieve_last_modified_date_time(db_object):
+    epoch = "1980-01-01T00:00:00.000Z"
+    last_modified_date_time = retrieve_date_time_element('_lastModifiedDateTime', db_object)
+    created_date_time = retrieve_date_time_element('createdDateTime', db_object)
+    if last_modified_date_time:
+        return last_modified_date_time
+    if created_date_time:
+        return created_date_time
+    return epoch
+
+def retrieve_date_time_element(key, db_object):
+    date_element = db_object.get(key)
+    if date_element is not None:
+        if isinstance(date_element, dict):
+            date_sub_element =  date_element.get('$date')
+            if date_sub_element is not None:
+                return date_sub_element
+        else:
+            return date_element
+    return ""
+
+def format_date_to_valid_outgoing_format(current_date_time):
+    parsed_date_time = get_valid_parsed_date_time(current_date_time)
+    parsed_date_time.astimezone(pytz.utc)
+    parsed_date_time = datetime.strftime(parsed_date_time, '%Y-%m-%dT%H:%M:%S.%f')
+    (dt, micro) = parsed_date_time.split('.')
+    dt = "%s.%03d%s" % (dt, int(micro) / 1000, 'Z')
+    return dt
+
+def get_valid_parsed_date_time(time_stamp_as_string):
+    valid_timestamps =  ['%Y-%m-%dT%H:%M:%S.%fZ','%Y-%m-%dT%H:%M:%S.%f%z']
+    for time_stamp_fmt in valid_timestamps:
+        try:
+            return datetime.strptime(time_stamp_as_string, time_stamp_fmt)
+        except Exception:
+            print(f'timestampAsString did not match valid format {time_stamp_fmt}')
+
+    # TODO Think about the below exception later on
+    raise Exception(f'Unparseable date found: {time_stamp_as_string} , did not match any supported date formats {valid_timestamps}')
+
+def sanitize(z):
+    decrypted = z['decrypted']
+    db_name = z['db_name']
+    collection_name = z['collection_name']
+    if ((db_name == "penalties-and-deductions" and collection_name == "sanction")
+            or (db_name == "core" and collection_name == "healthAndDisabilityDeclaration")
+            or (db_name == "accepted-data" and collection_name == "healthAndDisabilityCircumstances")):
+        decrypted = re.sub(r'(?<!\\)\\[r|n]', '', decrypted)
+    return decrypted.replace("$", "d_").replace("\u0000", "").replace("_archivedDateTime", "_removedDateTime").replace("_archived", "_removed")
```

Comment:
we shouldn't have hard-coded collecitons anymore

Reply:
Only these 3 collections data needs special massaging  Putting this in a db or configuration file can be an overkill . It has been implemented in the similar way in htme https://github.com/dwp/hbase-to-mongo-export/blob/0ed6e6999cb9e1d098536516cce4720bab4d0dac/src/main/kotlin/app/batch/processor/SanitisationProcessor.kt#L35